### PR TITLE
Fix errors when PRBs are enabled and products have empty prices

### LINKED ISSUE #4579: Product Variations without Prices Result in Critical Error on Product Page in PHP version 8.0
### Describe the bug

When a variable product or subscription is created with no prices added to any variation, the product page results in a Critical error in PHP version 8.0.

### To Reproduce

1. On an Atomic site, go to Settings > Hosting Config and set PHP to version 8.0
2. Create a variable product or subscription and leave all prices empty
3. Publish Product
4. Visit the product page

### Actual behavior

The product page displays a critical error.  In CLI, the full error log looks like this:

```
PHP Fatal error:  Uncaught TypeError: Unsupported operand types: string + int in /srv/htdocs/wp-content/plugins/woocommerce-payments/includes/class-wc-payments-payment-request-button-handler.php:295
Stack trace:
#0 /srv/htdocs/wp-content/plugins/woocommerce-payments/includes/class-wc-payments-payment-request-button-handler.php(656): WC_Payments_Payment_Request_Button_Handler->get_product_data()
#1 /wordpress/core/6.0.1/wp-includes/class-wp-hook.php(307): WC_Payments_Payment_Request_Button_Handler->scripts('')
#2 /wordpress/core/6.0.1/wp-includes/class-wp-hook.php(331): WP_Hook->apply_filters(NULL, Array)
#3 /wordpress/core/6.0.1/wp-includes/plugin.php(476): WP_Hook->do_action(Array)
#4 /wordpress/core/6.0.1/wp-includes/script-loader.php(2156): do_action('wp_enqueue_scri...')
#5 /wordpress/core/6.0.1/wp-includes/class-wp-hook.php(307): wp_enqueue_scripts('')
#6 /wordpress/core/6.0.1/wp-includes/class-wp-hook.php(331): WP_Hook->apply_filters(NULL, Array)
#7 /wordpress/core/6.0.1/wp-includes/plugin.php(476): WP_Hook->do_action(Array)
#8 /wordpress/core/6.0.1/wp-includes/general-template.php(3042): do_action('wp_head')
#9 /wordpress/core/6.0.1/wp-includes/template-canvas.php(17): wp_head()
#10 /wordpress/core/6.0.1/wp-includes/template-loader.php(106): include('/wordpress/core...')
#11 /wordpress/core/6.0.1/wp-blog-header.php(19): require_once('/wordpress/core...')
#12 /wordpress/core/6.0.1/index.php(17): require('/wordpress/core...')
#13 {main}
  thrown in /srv/htdocs/wp-content/plugins/woocommerce-payments/includes/class-wc-payments-payment-request-button-handler.php on line 295
 
```

### Expected behavior

For the product page to load normally despite not having prices for variations set.

### Workarounds

Adding prices to all variations or downgrading to PHP version 7.4 resolves the error.  

### Relevant links

Slack: p1660540852249059-slack-C03TY6J1A

### PR DESCRIPTION
**Pull request by a-danae**
Fixes #4579

#### Changes proposed in this Pull Request

Make products with an empty price unsupported for PRB.

Adding a check under `is_product_supported` to exclude PRB from products with no price.

#### Testing instructions

- Make sure you're using a site with PHP 8. Jurassic Ninja and ephemeral sites do the trick

**Variable products**
- As a merchant, create a variable product with a couple of variations
- Leave the prices of all variations empty
- As a shopper, go to the product page you just created
- Confirm the page loads fine and there are no errors

**Subscriptions**
- As a merchant, create a simple subscription with an empty subscription price
- As a shopper, go to the product page you just created
- Confirm the page loads fine and there are no errors

**Regression** 
- Confirm paying with PRBs continue to work fine

Please test other scenarios you see that could be impacted by this change, or that still be problematic.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Review by ricardo — APPROVED** [review]
LGTM and tests well.

The check in `is_product_supported` prevents further operations with an empty string `''`.

File: includes/class-wc-payments-payment-request-button-handler.php [hunk 1]
```diff
@@ -220,7 +220,7 @@ public function get_product_price( $product ) {
 		}
 
 		// Add subscription sign-up fees to product price.
-		if ( 'subscription' === $product->get_type() && class_exists( 'WC_Subscriptions_Product' ) ) {
+		if ( 'subscription' === $product->get_type() && class_exists( 'WC_Subscriptions_Product' ) && is_numeric( $product_price ) ) {
```
Comment (naman03malhotra) [inline includes/class-wc-payments-payment-request-button-handler.php:223]: We should return a default value for `$product_price` if both the conditions in this function are false.

Comment (naman03malhotra) [reply includes/class-wc-payments-payment-request-button-handler.php:223]: `get_price()` function is something to be careful with, it has different values for different cases, see https://github.com/Automattic/woocommerce-payments/pull/1774#discussion_r634867263 @jessy-p is creating an issue to improve documentation around it.

**Review by deepakpathania — APPROVED** [review]
Tests well and changes LGTM.

File: includes/class-wc-payments-payment-request-button-handler.php (hunk 1 above)
Comment (naman03malhotra) [reply includes/class-wc-payments-payment-request-button-handler.php:223]: FYI: If the product has no price, it returns an empty string and an integer if it has a value.

**Review by naman03malhotra — APPROVED** [review]
Tests well, the current solution works.
Thanks for working on it @a-danae

**Pull request merged**